archive/r/ruby/file-io.rb

```diff
@@ -0,0 +1,23 @@
+def write_file
```

Comment:
It will probably be fine this time(™), but next time you should really make sure to base your work on the latest version of the master branch of TheRenegadeCoder/sample-programs.

Reply:
I was trying to sync my fork but I kept getting tons of merge conflicts so I might have accidentally messed something up.

Reply:
Since you're using VS Code, there's really no way for me to help you (I'm not familiar with it at all). But like I said, the concepts should be pretty much the same. Take a look at [this guide](https://github.com/LambdaSchool/CS-Wiki/wiki/Keeping-GitHub-Forks-in-Sync); it's pretty much the same thing I do but described much better than I could.

Reply:
I ran all the commands I was suppose to and I got this:

```
➜  sample-programs master ✓ git remote -v
origin  https://github.com/Noah11012/sample-programs.git (fetch)
origin  https://github.com/Noah11012/sample-programs.git (push)
upstream        https://github.com/TheRenegadeCoder/sample-programs (fetch)
upstream        https://github.com/TheRenegadeCoder/sample-programs (push)
➜  sample-programs master ✓ git fetch upstream
➜  sample-programs master ✓ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
➜  sample-programs master ✓ git branch
➜  sample-programs master ✓ git merge upstream/master
Already up to date.
➜  sample-programs master ✓ git push
Everything up-to-date
➜  sample-programs master ✓
```

Reply:
But on my fork on github it says I have 15 commits ahead of the original repo.

Reply:
**Assuming you don't want to keep the commits that are on your master branch but not on the upstream master branch (you can view those [here](https://github.com/TheRenegadeCoder/sample-programs/compare/master...Noah11012:master) and decide for yourself if they're important)**, can you try doing:
```shell
$ git checkout master
$ git fetch upstream
$ git reset --hard upstream/master
$ git push --force
```

Reply:
Thank you! My fork is now synced.

Reply:
I'll make sure to sync with the original repo every time it gets updated.

Reply:
Looking good :+1:

(Btw, the command sequence you posted is what you should be using from now on, `reset --hard` and `push --force` are only ever to be used as a last measure)

Reply:
Will keep that in mind.